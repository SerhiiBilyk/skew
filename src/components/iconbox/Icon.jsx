import React from 'react';
import ReactDOM from 'react-dom';
import styles from './icon.scss';
import CSSModules from 'react-css-modules';
import PropTypes from 'prop-types';



class Icon extends React.PureComponent{
  constructor(props){
    super(props);
  }
  render(){
    var {theme,size,icon}=this.props;
    return(
      <div styleName={`circle ${this.props.theme} ${this.props.size}`}>
          <span className={`fa fa-${this.props.icon} fa-2x`}></span>
      </div>
    )
  }
}

Icon.propTypes = {
  theme: PropTypes.oneOf(['black','white']),
  size: PropTypes.oneOf(['big','middle','small']),
  icon: PropTypes.string.isRequired
}
export default CSSModules(Icon, styles, {allowMultiple: true})
