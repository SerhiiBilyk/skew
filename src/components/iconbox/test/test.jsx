import React from 'react';
import Icon from '../Icon.jsx';
import  {shallow, mount} from 'enzyme';
import renderer from 'react-test-renderer';


/*
TODO
why some tests dont include next 3 lines of code and work correctly
and some tests include this lines because works wrong
*/



/**SNAPSHOT*/
it('renders correctly', () => {
  const tree = renderer.create(
   <Icon icon='heart' theme='black' size='big'/>
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
