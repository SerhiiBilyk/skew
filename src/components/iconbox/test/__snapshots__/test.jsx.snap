// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="circle black big"
>
  <span
    className="fa fa-heart fa-2x"
  />
</div>
`;
